### ISSUE TITLE
Issue with go generate when using the privacy feature flag

### ISSUE THREAD

**Issue by adayNU** [facebook/ent#1137]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/facebook/ent/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

We are using the privacy feature flag which is great. We began using this initially in development by simply adding
`--feature privacy` to our `go generate` directive (and running it). We then built mixins on top of this. Everything is still grand (i.e. `go generate` still works fine).

We are not committing generated code to our repo. (I.e. we only have `ent/generate.go` and `ent/schema/*`) in source. When our CI/CD tries to run `go generate` its throwing an error:

```
... ent/schema/customer.go:4:2: could not import github.com/fabiustech/workspace/pkg/ent/privacy (invalid package name: "")
```

(Note: `ent/schema/customer.go` does indeed import the `privacy` package, as we've defined some `Policy`s there).

TL;DR - it seems like if you don't run `go generate` with the `--feature privacy` flag prior to any code existing that depends on it, `go generate` will fail.

## Expected Behavior 🤔

The `go generate` directive succeeds as if I had run it before writing any code dependent on the `privacy` package.

## Steps to Reproduce 🕹

I've reproduced it in [this repo](https://github.com/adayNU/ent-bug).

Steps:

1. Clone the repo.
2. run `go generate ./...` from the repo root.

## Your Environment 🌎

See `go.mod` in the repo.

**Comment by a8m** [comment]
Hey @adayNU!

> We are not committing generated code to our repo. (I.e. we only have ent/generate.go and ent/schema/*) in source. When our CI/CD tries to run go generate its throwing an error.

Sure, because the schema package depends on the generated privacy package, and ent can't build the schema for loading it.

Someone just asked in the issue-tracker if they should keep generated code in VCS, and I recommended to do it, because this is how I work and have experience with it (see https://github.com/facebook/ent/issues/1082).  

Having said all that, the issue can be solved using the [`schema/snapshot` feature-flag](https://entgo.io/docs/feature-flags/#auto-solve-merge-conflicts). This feature-flag tells ent to store a JSON version of the schema in an `internal` package, and in cases of build errors from the Go compiler (merge-conflicts or code doesn't exist - like yours), it uses this version to recover.

Closing, as this should solve your problem. Please feel free to re-open if you still have troubles with it. Thanks!

**Comment by adayNU** [comment]
👍  we'll move to keeping it in source then. After digging a bit into the generated code in the privacy package this makes a lot more sense to me :)
